// Copyright (c) 2022 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;

option java_outer_classname = "WorldObjectServiceProto";

import "bosdyn/api/world_object.proto";

// The world object service provides a way to track and store objects detected in the world around the robot.
service WorldObjectService {
    // Request a list of all the world objects in the robot's perception scene.
    rpc ListWorldObjects(ListWorldObjectRequest) returns (ListWorldObjectResponse) {}

    // Mutate (add, change, or delete) the world objects.
    rpc MutateWorldObjects(MutateWorldObjectRequest) returns (MutateWorldObjectResponse) {}
}
